File: SeLeAs/Python/maximumNumber.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        vowels = {'a', 'e', 'i', 'o', 'u'}
        curr = 0
        
        # Initial window of size k
        for i in range(k):
            if s[i] in vowels:
                curr += 1
        
        max_vowels = curr  # Initialize max_vowels after the first window
        
        # Sliding window approach
        for i in range(k, len(s)):
            if s[i] in vowels:
                curr += 1
            if s[i - k] in vowels:
                curr -= 1
            
            # Update the max_vowels with the maximum count found so far
            max_vowels = max(max_vowels, curr)
        
        return max_vowels

File: SeLeAs/LeetCode/LeetCode_75/maximumNumber.java
class maximumNumber {
    public int maxVowels(String s, int k) {
        // Set of vowels for quick lookup
        String vowels = "aeiou";
        int curr = 0;
        
        // Initial window of size k
        for (int i = 0; i < k; i++) {
            if (vowels.indexOf(s.charAt(i)) != -1) {  // Check if character is a vowel
                curr++;
            }
        }
        
        int maxVowels = curr;  // Initialize max_vowels after the first window
        
        // Sliding window approach
        for (int i = k; i < s.length(); i++) {
            // Add the new character to the window
            if (vowels.indexOf(s.charAt(i)) != -1) {
                curr++;
            }
            
            // Remove the character that is sliding out of the window
            if (vowels.indexOf(s.charAt(i - k)) != -1) {
                curr--;
            }
            
            // Update maxVowels with the maximum count found so far
            maxVowels = Math.max(maxVowels, curr);
        }
        
        return maxVowels;
    }
}
